### DISCUSSION TITLE
Argocd could not look for path in git repo

### DISCUSSION THREAD

**Discussion by xcloudx87** [argoproj/argo-cd#6970] [Q&A] [answered]
Hi there,

I am having strange issue when using argocd. Below is my command to install argocd:

> helm install argocd argo/argo-cd -n argocd --set fullnameOverride=argocd --set installCRDs=false --set server.extraArgs={--insecure} --set dex.enabled=false --set server.config.url="http://argocd.global.com"

With domain argocd.global.com, ArgoCD could not look for helm chart automatically when i add application.
![image](https://user-images.githubusercontent.com/71874570/129175735-11932e6c-ee8f-4254-b702-fbd3f932ab93.png)

But if i forward port from k8s and access localhost:8080 it works fine
![image](https://user-images.githubusercontent.com/71874570/129175480-e625362f-7eb4-48b9-8eb8-210710715189.png)

I don't know where i got wrong. 

Could you please advise?

Thanks in advance!

**Comment by jgwest** [comment, marked as answer]
Hi @xcloudx87,

The helm chart doesn't include an Ingress, so if you want to expose and access the Argo CD web ui (without using port forwarding) you would need to follow these instructions: https://argoproj.github.io/argo-cd/operator-manual/ingress/#kubernetesingress-nginx (if you haven't already). If you have, you may want to check your setup, as improper ingress configuration can cause these types of issues.

You may also want to try removing `--set server.config.url="http://argocd.global.com"` which, according to the Helm values file is optional, so you may want to remove it and see if that changes the behaviour.

My guess is that if you open the developer console in your browser (F12), and select Console, or select Network, you may see HTTP errors in connecting to the web UI (in the failing case, when you visit the page you screenshotted), which would be hopefully be solved by following the above ingress instructions.

Finally, it couldn't hurt to check the Argo CD pod logs for any obvious errors, especially server and repo server.

**Reply by xcloudx87** [reply]
Hi @jgwest,

Thank you very much for your response. I have removed `--set server.config.url="http://argocd.global.com"` from installation command and no luck. 

For Ingress, currently my topology is client -> Nginx reverse proxy -> Kong API -> ArgoCD service. Where Nginx keep domain pointing with IP in DNS, and Kong's IP is bind with nginx's upstream. So I think I will not need Ingress.

I have looked at F12 at Console and noticed that there was an error (404 error) while calling 

> http://argocd.global.com/api/v1/repositories/http%3A%2F%2Fgit.cpn.com%2Fxrepo%2Fargo-cd.git/appdetails

It works fine with localhost:8080, and there was no clue from ArgoCD pod logs. 

Could you please advise?

Thanks in advance!

**Reply by jgwest** [reply]
Hi @xcloudx87, not much more I can suggest I think, beyond checking that your topology is correctly redirecting requests consistent with what Argo CD is expecting (described in the doc above). 

If anyone else is using a similar topology and has any suggestions, feel free to reply.

**Reply by xcloudx87** [reply]
Hi @jgwest,

After give a quick refactor for toplogy, i found that ArgoCD didn't work with simpler domain pointing topology. I have just exposed service argo to NodePort, remove Kong from topology and it still doesn't work. Here is my config for nginx reverse proxy:

> server {
>     listen 80;
>     server_name argocd.global.com;
> 
>     access_log /u01/logs/nginx/argocd.access.log;
>     error_log /u01/logs/nginx/argocd.error.log;
> 
>     # allow large uploads of files
>     client_max_body_size 100M;
> 
>     location / {
>             proxy_pass http://192.168.1.5:32401/;
>             proxy_set_header Host $host;
>             proxy_set_header X-Real-IP $remote_addr;
>             proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>     }
> }

Maybe i will need to move to the issues tab? Could you please advise?

**Reply by xcloudx87** [reply]
Hi @jgwest,

Finally i made it work. It seems like my wrong in configuration with Nginx. I have removed trailing slash at proxy_pass on Nginx side and it works fine now.
So before:

> location / {
>         **proxy_pass http://192.168.1.5:32401/;**
>         proxy_set_header Host $host;
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> }

And after:

> location / {
>         **proxy_pass http://192.168.1.5:32401;**
>         proxy_set_header Host $host;
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> }

Thank you very much for your time!
